### PR TITLE
raise clear error for module not available

### LINKED ISSUE #1422: Raise clearer error when module not available
Context:

> Hi, @jayqi - Thank you for the question. In terms of providing a more clear error, this hasn't been fixed. I'll re-open this issue, and I think a refactor will be good in Flash to add better errors if the module is not found. In case you are interested to pick this up, please feel free (also let me know so that I can assign you the issue). Else, I'll try to scope this for the next week, and create an issue and a PR around this. 🎉

cc: @ethanwharris - What do you think about the refactor to raise errors instead of running into: `NameError "T" not defined``.

Hint: https://github.com/Lightning-AI/lightning-flash/issues/1403#issuecomment-1219792770

_Originally posted by @krshrimali in https://github.com/Lightning-AI/lightning-flash/issues/1403#issuecomment-1219780935_

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

Fixes #1422 #1423. This PR:

* Adds `@requires("kornia")` decorator to image segmentation input transform code for better coverage and clearer error.
* Since Audio Classification uses `torchvision`, a `requires` decorator for torchvision is also added in the relevant file.

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [x] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by krshrimali — APPROVED** [review]
Thank you, @aniketmaurya for this PR! LGTM 🔥 A suggestion and a question:

1. Let's also add `torchvision` requirement to [`datatype_audio.txt`](https://github.com/Lightning-AI/lightning-flash/blob/master/requirements/datatype_audio.txt) file, since we need it for audio classification. Can you please do this in this PR?
2. Question: just cross-checking, did you make sure that a clearer error is raised if kornia is not installed and a user attempts to run this example?

Also, I've modified the PR description for future reference, just in case anyone hops on this for any reason! :) Hope that sounds good to you.

**Comment by aniketmaurya** [comment]
Thank you for reviewing the PR and suggestions @krshrimali 😄 ! I have added `torchvision` to audio requirements and also cross-checked the error message when Kornia is not uninstalled. The error message clearly states that Konia is not installed 👇🏻 

`ModuleNotFoundError: Required dependencies not available. Please run: pip install 'kornia'`

**Comment by krshrimali** [comment]
@aniketmaurya - Sounds great, thank you so much! However, I can't see any commit after your comment. Did you forget to push the changes for requirements? ;)

**Comment by aniketmaurya** [comment]
Hi @krshrimali, yeah sorry I forgot to push it earlier. I've updated the PR just now! 😅

**Comment by krshrimali** [comment]
Update: added CHANGELOG entries! cc: @ethanwharris - if you can merge this whenever you are back! :)

File: flash/audio/classification/input_transform.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def train_per_sample_transform(self) -> Callable:
             ]
         )
 
+    @requires("torchvision")
```
Comment (ethanwharris) [inline flash/audio/classification/input_transform.py:54]:
```suggestion
    @requires("audio")
```

File: flash/image/segmentation/input_transform.py [hunk 2]
```diff
@@ -47,6 +47,7 @@ class SemanticSegmentationInputTransform(InputTransform):
     mean: Union[float, Tuple[float, float, float]] = (0.485, 0.456, 0.406)
     std: Union[float, Tuple[float, float, float]] = (0.229, 0.224, 0.225)
 
+    @requires("kornia")
```
Comment (ethanwharris) [inline flash/image/segmentation/input_transform.py:50]:
```suggestion
    @requires("image")
```

File: flash/image/segmentation/input_transform.py [hunk 3]
```diff
@@ -61,6 +62,7 @@ def train_per_sample_transform(self) -> Callable:
             ]
         )
 
+    @requires("kornia")
```
Comment (ethanwharris) [inline flash/image/segmentation/input_transform.py:65]:
```suggestion
    @requires("image")
```

File: flash/image/segmentation/input_transform.py [hunk 4]
```diff
@@ -72,6 +74,7 @@ def per_sample_transform(self) -> Callable:
...
```
Comment (ethanwharris) [inline flash/image/segmentation/input_transform.py:77]:
```suggestion
    @requires("image")
```

**Pull request merged**